casanovo/data/ms_io.py

```
@@ -14,6 +14,51 @@
+    -------
+    dict
+        Mapping from spectrum index (as a string, e.g. ``"0"``) to the
+        SCANS field value (e.g. ``"17"``). Only spectra that contain a
+        ``SCANS``, ``SCAN``, or ``SCAN ID`` header field are included.
+    """
+    index2scan = {}
+    index = 0
+    current_scan = None
+    in_ions = False
+    try:
+        with open(mgf_path, "r", errors="replace") as fh:
```

Ruff (0.15.6)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)